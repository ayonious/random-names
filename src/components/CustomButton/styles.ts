import styled from 'styled-components';

export const CenteredFlexDevWrapper = styled.div`
  display: flex;
  align-content: center;
  justify-content: center;
`;
