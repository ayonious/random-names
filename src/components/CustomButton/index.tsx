import Button from '@material-ui/core/Button';
import React from 'react';

import { CenteredFlexDevWrapper } from './styles';

interface Props {
  title: string;
  onClick: () => void;
}

const CustomButton = (props: Props) => {
  return (
    <CenteredFlexDevWrapper>
      <Button variant="contained" color="primary" onClick={props.onClick}>
        {props.title}
      </Button>
    </CenteredFlexDevWrapper>
  );
};

export default CustomButton;
