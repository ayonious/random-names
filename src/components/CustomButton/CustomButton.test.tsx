/**
 * @jest-environment jsdom
 */
import { expect } from '@jest/globals';
import { render } from '@testing-library/react';
import React from 'react';

import CustomButton from '.';

it('CustomButton renders', () => {
  const func = () => {
    console.log('on click function testing');
  };

  const { container } = render(
    <CustomButton onClick={func} title={'Shuffle'} />
  );
  expect(container).toMatchSnapshot();
});
