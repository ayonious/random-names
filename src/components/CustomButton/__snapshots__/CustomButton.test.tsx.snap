// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomButton renders 1`] = `
<div>
  <div
    class="sc-bdvuGq kSiELm"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
      data-testid="shuffle-button"
      tabindex="0"
      type="button"
    >
      Shuffle
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
